src/XRP/issued-currency-client.ts

```diff
@@ -1024,4 +1033,107 @@ export default class IssuedCurrencyClient {
     // If we have too many digits of precision, express with scientific notation but fit within decimal precision.
     return rawSendMaxValue.toExponential(14, 0).replace(/e\+/, 'e')
   }
+
+  /**
+   * Creates an offer on the XRP Ledger.
+   * @see https://xrpl.org/offers.html
+   * @see https://xrpl.org/offercreate.html#offercreate
+   *
+   * @param sender The Wallet creating this offer, and that will sign the transaction.
+   * @param takerGetsAmount The amount and type of currency being provided by the offer creator.
+   *               Use a string to specify XRP, or an IssuedCurrency object to specify an issued currency.
+   * @param takerPaysAmount The amount and type of currency being requested by the offer creator.
+   *               Use a string to specify XRP, or an IssuedCurrency object to specify an issued currency.
+   * @param offerSequence (Optional) An offer to delete first, specified by the sequence number of a previous OfferCreate transaction.
+   *               If specified, cancel any offer object in the ledger that was created by that transaction.
+   *               It is not considered an error if the offer specified does not exist.
+   * @param expiration (Optional) Time after which the offer is no longer active, in seconds since the Ripple Epoch.
+   *               (See https://xrpl.org/basic-data-types.html#specifying-time)
+   */
+  public async createOffer(
+    sender: Wallet,
+    takerGetsAmount: string | IssuedCurrency,
+    takerPaysAmount: string | IssuedCurrency,
+    offerSequence?: number,
+    expiration?: number,
+  ): Promise<TransactionResult> {
+    // construct TakerGets
+    const takerGetsCurrencyAmount = new CurrencyAmount()
```

Comment:
this is super cool, but i think it may be better to have this be a `string` to start, since it matches the rest of our apis. we don't rly seem to have too many apis that use protobuffs in this way (i guess we could though?)

 it's also a bit less risky to start with a more opinionated/restrictive api and make it accept more later...would strongly recommend keeping this to be just a string to start and expanding later if we like the approach of having a protobuf passed in

Reply:
hmmm, I'm not sure what you mean here?  it can't just be a string because that's how you specify an XRP amount, but we have to be able to specify both an XRP amount AND an issued currency amount (especially bc if either one is an XRP amount, the other one has to be issued currency (otherwise it's not an issued currency).  the opposite isn't true, since there could be two different issued currencies specified).  Also, here an `IssuedCurrency` isn't a protobuf, it's an interface that we've defined to bundle the three piece of information together.  You're right that our other APIs actually don't take `IssuedCurrencies` and instead accept the information piecemeal... something that we have an asana task out to change, bc this is much more logical.